src/hdf5_time_series_storage.jl

```diff
@@ -300,24 +327,121 @@ function deserialize_time_series(
         path = _get_time_series_path(root, uuid)
         attributes = _read_time_series_attributes(storage, path, rows, T)
         @assert attributes["type"] == T
-        @assert length(attributes["dataset_size"]) == 2
         @debug "deserializing a Forecast" T
-        data = SortedDict{Dates.DateTime, Array}()
-        initial_timestamp = attributes["start_time"]
-        interval = attributes["interval"]
-        start_time = initial_timestamp + interval * (columns.start - 1)
-        if length(columns) == 1
-            data[start_time] = path["data"][rows, columns.start]
-        else
-            data_read = path["data"][rows, columns]
-            for (i, it) in
-                enumerate(range(start_time; length = length(columns), step = interval))
-                data[it] = @view data_read[1:length(rows), i]
-            end
+        data_type = attributes["data_type"]
+        data = get_hdf_array(path["data"], data_type, attributes, rows, columns)
+        new_ts = T(ts_metadata, data)
+    end
+end
+
+function get_hdf_array(
+    dataset,
+    ::Type{<:CONSTANT},
+    attributes::Dict{String, Any},
+    rows::UnitRange{Int},
+    columns::UnitRange{Int},
+)
+    data = SortedDict{Dates.DateTime, Array}()
```

Comment:
we can do this too. 

```suggestion
function get_hdf_array(
    dataset,
    ::Type{<:T},
    attributes::Dict{String, Any},
    rows::UnitRange{Int},
    columns::UnitRange{Int},
) where T <: CONSTANT
    data = SortedDict{Dates.DateTime, Array{T}}()
```

Reply:
if I make this change I get a error message saying T not defined when creating the SortedDict

Reply:
Then let's leave it like it is, interesting error. I got similar strange things with sorted dicts.